rest_lights.cpp

```diff
@@ -1974,101 +1974,135 @@ int DeRestPluginPrivate::setWindowCoveringState(const ApiRequest &req, ApiRespon
 int DeRestPluginPrivate::setTuyaDeviceState(const ApiRequest &req, ApiResponse &rsp, TaskItem &taskRef, QVariantMap &map)
 {
     QString id = req.path[3];
+    
     bool targetOn = false;
-
-    if (map.contains("on"))
+    bool hasOn = false;
+    bool hasBri = false;
+    uint targetBri = 0;
+    
+    bool ok = false;
+    
+    //Parse all parameters
+    for (QVariantMap::const_iterator p = map.begin(); p != map.end(); p++)
     {
-        if (map["on"].type() == QVariant::Bool)
+        if (p.key() == "bri" && R_GetProductId(taskRef.lightNode).startsWith(QLatin1String("Tuya_DIMSWITCH")))
         {
-            bool ok = false;
-            qint8 button = DP_IDENTIFIER_BUTTON_1;
-            QByteArray data;
-
-            targetOn = map["on"].toBool();
-
-            //Retreive Fake endpoint, and change button value
-            uint8_t ep = taskRef.lightNode->haEndpoint().endpoint();
-            if (ep == 0x02) { button = DP_IDENTIFIER_BUTTON_2; }
-            if (ep == 0x03) { button = DP_IDENTIFIER_BUTTON_3; }
-
-            //Use only the first endpoint for command
-            taskRef.req.setDstEndpoint(0x01);
+            if (map[p.key()].type() == QVariant::Double)
+            {
+                targetBri = map["bri"].toUInt(&ok);
+                if (ok)
+                {
+                    hasBri = true;
+                }
+            }
 
-            DBG_Printf(DBG_INFO, "Tuya debug 10: EP: %d ID : %s\n", ep, qPrintable(id));
+            if (!hasBri)
+            {
+                rsp.list.append(errorToMap(ERR_INVALID_VALUE, QString("/lights/%1").arg(id), QString("invalid value, %1, for parameter, bri").arg(map["bri"].toString())));
+            }
+        }
 
-            if (targetOn)
+        else if (p.key() == "on" && taskRef.lightNode->item(RStateOn))
+        {
+            if (map[p.key()].type() == QVariant::Bool)
             {
-                data = QByteArray("\x01", 1);
+                hasOn = true;
+                targetOn = map["on"].toBool();
             }
             else
             {
-                data = QByteArray("\x00", 1);
+                rsp.list.append(errorToMap(ERR_INVALID_VALUE, QString("/lights/%1/state").arg(id), QString("invalid value, %1, for parameter, on").arg(map["on"].toString())));
             }
+        }
 
-            ok = sendTuyaRequest(taskRef, TaskTuyaRequest, DP_TYPE_BOOL, button, data);
+        else
+        {
+            rsp.list.append(errorToMap(ERR_PARAMETER_NOT_AVAILABLE, QString("/lights/%1/state").arg(id), QString("parameter, %1, not available").arg(p.key())));
+        }
+    }
 
-            if (ok)
+    if (hasBri)
+    {
+        if (targetBri <= 0xFF)

```

Comment:
With the above change the `if (targetBri <= 0xFF)` can be removed, since `hasBri` also means a valid value.

Reply:
done